devicetypes/smartthings/fidure-thermostat.src/fidure-thermostat.groovy

```diff
@@ -292,7 +292,7 @@ def parseDescriptionAsMap(description) {
     def result = map.get('result')
     def list = [];
 
-  if (getDataLengthByType(map.get('encoding')) != null && getDataLengthByType(map.get('encoding')) < map.get('value').length()) {
+  if (encoding != null && getDataLengthByType(encoding) != null && getDataLengthByType(encoding) < value.length()) {
```

Comment:
Could fix the messed up indentation in this method while you're in here if you want

Reply:
there's so much that is so bad I'm scared to touch it.

Reply:
The file definitely has a case of mixed indentation... The memories of using the IDE and submitting through it...

I'd probably just do the whole file at once, and probably not as part of this PR.